### PR TITLE
Add fix for no-hbac-allow option in server install

### PR DESCRIPTION
**Pull request by Akasurde**
Fixes: https://fedorahosted.org/freeipa/ticket/6357

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

### CODE REVIEW

**Comment by tomaskrizek** [comment]
`--no_hbac_allow` option should remain as an undocumented option for backwards compatibility.

**Comment by Akasurde** [comment]
@tomaskrizek Should I remove 'no_hbac_allow' option from ipa-server-install man page then ?

**Comment by tomaskrizek** [comment]
@Akasurde The man page is correct. However, we can no longer use the command 
```
ipa-server-install --no_hbac_allow
```
if you simply rename the option. We need to keep this option for backwards compatibility and add a new option `--no-hbac-allow`. Internally, `no_hbac_allow` should only be an alias to `no-hbac-allow`. As a result, both of the following commands should work:
```
ipa-server-install --no-hbac-allow
ipa-server-install --no_hbac_allow
```
With `--no-hbac-allow` being the preferred and documented option, while `--no_hbac_allow` simply remains for backwards compatibility. 

I think the correct way to fix the issue is to add a new option and then make sure both the options have the same effect when used.

File: ipaserver/install/server/install.py [hunk 1]
```diff
@@ -1294,7 +1294,7 @@ def idmax(self):
     no_hbac_allow = Knob(
         bool, False,
         description="Don't install allow_all HBAC rule",
-        cli_name='no_hbac_allow',
+        cli_name='no-hbac-allow',
```
Comment (jcholast) [inline ipaserver/install/server/install.py]: You should also add `cli_aliases=['no_hbac_allow']` for backward compatibility.

Comment (Akasurde) [inline ipaserver/install/server/install.py]: @jcholast Do I need to run `makeapi` too ?

**Comment by tomaskrizek** [comment]
LGTM, thanks for the patch!

**Comment by Akasurde** [comment]
@tomaskrizek @stlaz @jcholast Thanks for review comments.

**Comment by mbasti-rh** [comment]
Fixed upstream
master:
https://fedorahosted.org/freeipa/changeset/a42059228018839ae2656c27f5b00d96bc935ee3

**Pull request closed without merging**